### PR TITLE
Allow to download the GeoNames data at post-install

### PR DESCRIPTION
**Pull request by itaisatati**
My second PR for this project :)

Addressing the suggestion at #53, this PR includes the following:
* A post-install script which runs the initialization function.
The parameters are passed via env-variables.
* A short description & example in the documentation
* All of the methods used here are supported from Node v6+, no breaking changes

Please let me know what you think:
* The files are tagged with the current date so initializing at a different date will not gain anything from this effort
Should there be an initialization option to save the files without a date tag?
* Should the `Dockerfile` be modified to use this post-install script instead of `curl`?
It will thereby get more usage and better feedback

### CODE REVIEW

File: postinstall.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        admin1: GEOCODER_POSTINSTALL_ADMIN1?.toLowerCase() === 'true',
+        admin2: GEOCODER_POSTINSTALL_ADMIN2?.toLowerCase() === 'true',
+        admin3And4: GEOCODER_POSTINSTALL_ADMIN3_AND_4?.toLowerCase() === 'true',
+        alternateNames:
+          GEOCODER_POSTINSTALL_ALTERNATE_NAMES?.toLowerCase() === 'true',
+      },
+      countries: GEOCODER_POSTINSTALL_COUNTRIES?.split(',') || [],
+    },
+    function () {
+      console.info('[local-reverse-geocoder] post-install: Finished');
```
Comment (tomayac) [inline postinstall.js:44]:
```suggestion
      console.info('[local-reverse-geocoder] post-install: Finished.');
```

File: postinstall.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      process.exit(0);
+    }
+  );
+} catch (error) {
+  if (GEOCODER_POSTINSTALL_FAIL_SILENTLY?.toLowerCase() === 'true') {
+    console.warn(
+      '[local-reverse-geocoder] post-install: An error occurred. Detected fail-silently flag',
```
Comment (tomayac) [inline postinstall.js:51]:
```suggestion
      '[local-reverse-geocoder] post-install: An error occurred. Detected fail-silently flag.',
```

File: postinstall.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      error
+    );
+    process.exit(0);
+  } else {
+    console.error(
+      '[local-reverse-geocoder] post-install: An error occurred',
```
Comment (tomayac) [inline postinstall.js:57]:
```suggestion
      '[local-reverse-geocoder] post-install: An error occurred.',
```

File: README.md [hunk 4]
```diff
@@ -319,12 +319,16 @@ like Google's
...
-code into the array, it will disable geocoder from downloading all ~2GB(!) worth
-of data and only load the specified countries' data. If you want to re-enable
-geocoder to download all data, the countries array needs to be empty.
+of disk space). All follow-up requests are lightning fast.
+
+### Downloading specific sets of countries
+
+To reduce the time taken to initialize the data, you can manually configure it
+to only download a specific set of countries from Geonames. Do note that when
+you add a country code into the array, it will disable geocoder from downloading
```
Comment (tomayac) [inline README.md:328]:
```suggestion
you add a country code into the array, it will disable the geocoder from downloading
```

File: README.md [hunk 5]
```diff
@@ -319,12 +319,16 @@ like Google's
...
+all ~2GB(!) worth of data and only load the specified countries' data. If you
+want to re-enable geocoder to download all data, the countries array needs to be
```
Comment (tomayac) [inline README.md:330]:
```suggestion
want to re-enable the geocoder to download all data, the countries array needs to be
```

File: README.md [hunk 6]
```diff
@@ -347,6 +351,26 @@ geocoder.init(
...
+
+There's also the option of downloading the Geonames files via a post-install
+script.  
+The script is invoked automatically after installation but won't download any
+files without getting at least one of the init options in an env variable.  
+The options should be specific with a `GEOCODER_POSTINSTALL_` prefix.
+
+#### Example of downloading the files via the post-install script
+
+```javascript
```
[the remark this answers is not in the document]
Comment (itaisatati) [reply README.md]: I accepted the suggestion but notice that there are more of these in the README file. Might be worth modifying in the merge commit.

**Comment by itaisatati** [comment]
Thanks for reviewing :)
Let me know if there's something that's not in the issues tab and worth working on.

**Review by tomayac — APPROVED** [review]
LGTM, thank you!

**Comment by tomayac** [comment]
Just published as `v0.12.6`.

**Comment by ngoclong19** [comment]
`npm install` is failing with this error 'Error: Cannot find module '[...]\node_modules\local-reverse-geocoder\postinstall.js''.

I think `postinstall.js` is needed to included in array `files` of `package.json`.

**Comment by itaisatati** [comment]
@ngoclong19 Sorry about that.
https://github.com/tomayac/local-reverse-geocoder/pull/61: I created a PR to fix the problem

**Pull request merged**